pwndbg/disasm/arm.py

```diff
@@ -48,24 +47,37 @@ def condition(self, instruction):
         C = value & (1 << 29)
         V = value & (1 << 28)
 
-        cc = {
```

Comment:
I'm wondering whether we should keep this one. I think the original is much more readable.

Maybe we can declare the dictionary, and then check the key exists with `in`, otherwise return None

Reply:
Generally if/elif is more efficient in Python :x 

I think the  readability doesn't hurt much but I'm okay with dropping this change if u feel its better to use a dict

Reply:
Is it? Dictionaries should be an O(1) lookup compared to iterating over each if/else branch which will take longer.

Reply:
They are but we are creating the whole dictionary here; i bet this must be more operations since u need to hash it all etc.

Reply:
But yea... if/elif is not great either:
```
In [1]: def foo(x):
   ...:     if x == 1:
   ...:         return 1
   ...:     elif x == 2:
   ...:         return 2
   ...:     elif x == 3:
   ...:         return 3
   ...:     return None
   ...:

In [2]:

In [2]: import dis

In [3]: dis.dis(foo)
  2           0 LOAD_FAST                0 (x)
              2 LOAD_CONST               1 (1)
              4 COMPARE_OP               2 (==)
              6 POP_JUMP_IF_FALSE       12

  3           8 LOAD_CONST               1 (1)
             10 RETURN_VALUE

  4     >>   12 LOAD_FAST                0 (x)
             14 LOAD_CONST               2 (2)
             16 COMPARE_OP               2 (==)
             18 POP_JUMP_IF_FALSE       24

  5          20 LOAD_CONST               2 (2)
             22 RETURN_VALUE

  6     >>   24 LOAD_FAST                0 (x)
             26 LOAD_CONST               3 (3)
             28 COMPARE_OP               2 (==)
             30 POP_JUMP_IF_FALSE       36

  7          32 LOAD_CONST               3 (3)
             34 RETURN_VALUE

  8     >>   36 LOAD_CONST               0 (None)
             38 RETURN_VALUE

In [4]:
```

Reply:
Removed this change, merging